Topic recognition S4601130

Shivangiii opened the pull request:
Hi Shakes and team,
Please review my work, any feedback would be highly appreciated.

**This branch contains implementation for segmenting ISIC images using Improved Unet Architecture**
**Submitted by- Sivangi Mund, s4601130**

The submission contains a model.py(python file) containing the implementation of the model, a DriverScript.ipynb(jupyter notebook) containing all the pre-processing, compilation and result visualisation steps, a readMe file containing all necessary details of the implementation and a resources folder containing supporting result plots and images.

This implementation successfully achieves a dice similarity of more than 0.8 as per the goal.

SiyuLiu0329 commented:
- Commit message should be more detailed
- Results and code appear valid for Easy Difficulty

Other Feedback
- Good group of resources 
- Could use some basic comments in `model.py`. This is an open source project and your code might be maintained and modified by others.

rexsw commented:
Code looks good for normal difficulty

Comments such as:
   #appending into list
    x.append(x_tmp)
Don’t add much to the readability of the code and such be left out, but comments in def model() are better. However def model() could use more comments as it is a complex function. 

Tensorflow has image resizing and reading functions so you don’t need to use numpy and PIL, and it’s always helpful to reduce the number of libraries used in your project.

shakes76 commented on recognition/46011303/model.py:
```diff
@@ -0,0 +1,91 @@
+import tensorflow
```
Code header blocks should be present on each script.

shakes76 commented:
## TF Usage
OK
## Good Practice (Design/Commenting)
Good design 
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage, Readme sparse
Module present
## Commit Log
OK commit messages 
Progressive commits used 
## Documentation
Description good and Comments OK
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
**Feedback required. Add more comments to driver.**
Notebooks have markdown sections for you to explain what's going on, so you should make use of it.
In the future, a more detailed pull request could've been done. When developer receiving the request is not known, the request could be rejected. Remember you could use/reuse the README info.

Shivangiii commented:
Hi Shakes and team,
Thank you for all your valuable reviews. I have incorporated certain changes into the files of this pull request as per shake's feedbacks-
1. Addressed the developer in the beginning of the pull request description.
2. Added author header blocks to both my model and driver script files
3.  Added comments using markdown section of the driver script jupyter notebook.

Please let me know if there is anything else that needs to be changed.
Thanks!
